src/difference.jl

```diff
@@ -0,0 +1,34 @@
+"""
+$(TYPEDEF)
+
+Represents a difference operator.
+
+# Fields
+$(FIELDS)
+
+# Examples
+
+```jldoctest
+julia> using Symbolics
+
+julia> @variables t;
+
+julia> D = Difference(t; dt=0.01)
+Difference(t; dt=0.01)
+```
+"""
+struct Difference <: Function
+    """Fixed Difference"""
+    t
+    dt
+    Difference(t; dt) = new(value(t), dt)
```

Comment:
Is it better to have the constructor as `Difference(t; d)` instead of `Difference(t; dt)`? 
We might have scenarios where is the term is `x` when `Difference(x; dt)` would look odd.
OR can we assume that the Difference is usually used w.r.t time `t`?

Reply:
This is fine. @ChrisRackauckas what do you think?

Reply:
`dt` is common enough in the ecosystem that it's probably better for uniformity.